components/collector/src/source_collectors/dependency_track/base.py

```diff
@@ -59,7 +59,7 @@ async def _get_project_uuids(self) -> dict[str, str]:
 
     async def _get_projects(self) -> AsyncIterator[DependencyTrackProject]:
         """Return the Dependency-Track projects."""
-        projects_api = URL(await self._api_url() + "/project")
+        projects_api = URL(await DependencyTrackBase._api_url(self) + "/project")
```

Comment:
Why was this needed?

Reply:
Because the code was wrong :-) 

If a subclass would override the `_api_url` method to, for example, access the `/issues` endpoint, this method would also call that overridden method and try to access `/issues/project`. 

It was just a coincidence we haven't triggered this accident waiting to happen so far: only `DependencyTrackSourceUpToDateness` overrides `_api_url` at the moment, but it doesn't call `_get_projects`. The other subclasses do call `_get_projects` but don't override `_api_url`. Pfew!